nifi-commons/nifi-web-client/src/main/java/org/apache/nifi/web/client/StandardHttpUriBuilder.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.web.client;
+
+import okhttp3.HttpUrl;
+import org.apache.nifi.web.client.api.HttpUriBuilder;
+
+import java.net.URI;
+import java.util.Objects;
+
+/**
+ * Standard HTTP URI Builder based on OkHttp HttpUrl
+ */
+public class StandardHttpUriBuilder implements HttpUriBuilder {
+    private final HttpUrl.Builder builder;
+
+    public StandardHttpUriBuilder() {
+        this.builder = new HttpUrl.Builder();
+    }
+
+    @Override
+    public URI build() {
+        final HttpUrl httpUrl = builder.build();
+        return httpUrl.uri();
+    }
+
+    @Override
+    public HttpUriBuilder scheme(final String scheme) {
+        Objects.requireNonNull(scheme, "Scheme required");
+        builder.scheme(scheme);
+        return this;
+    }
+
+    @Override
+    public HttpUriBuilder host(final String host) {
+        Objects.requireNonNull(host, "Host required");
+        builder.host(host);
+        return this;
+    }
+
+    @Override
+    public HttpUriBuilder port(int port) {
+        builder.port(port);
+        return this;
+    }
+
+    @Override
+    public HttpUriBuilder encodedPath(final String encodedPath) {
```

Comment:
Good questions!

Requiring the full path to be encoded helps make it clear the individual path segments must be pre-encoded when using this particular method.

Supporting a non-encoded path would require careful parsing, considering `/` characters at minimum, and then encoding the path segments.

It might be worth considering additional options in future implementations, but it seems better to start with these methods and add new ones when necessary.